contents/julia_101.md

```diff
@@ -423,55 +517,2108 @@ logarithm(10)
+"""
+)
+```
+
+#### `randn` {#sec:random_randn}
+
+`randn` follows the same general principle from `rand` but now it only returns numbers generated from standard normal distribution.
+The standard normal distribution is the normal distribution with mean 0 and standard deviation 1.
+The default type is `Float64` and it only allows for subtypes of `AbstractFloat` or `Complex`:
+
+```jl
+scob(
+"""
+randn()
+"""
+)
+```
+
+We can only specify the size:
+
+```jl
+sco(
+"""
+randn((2, 2))
+"""
+)
+```
+
+#### `seed!` {#sec:random_seed}
+
+To finish off the `Random` overview, let's talk about **reproducibility**.
+Often, we want to make something replicable.
+Meaning that, we want the random number generator to generate the same random sequence of numbers,
+despite paradoxical that might sound...
+We can do so with the `seed!` function.
+
+Let me show you an example of a `rand` that generates the same three numbers given a certain seed:
+
+```jl
+sco(
+"""
+Random.seed!(123)
+rand(3)
+"""
+)
+```
+
+```jl
+sco(
+"""
+Random.seed!(123)
+rand(3)
+"""
+)
+```
+
+Note that `seed!` is not automatically exported by the `Random` module.
+We have to call it with the `Module.function` syntax.
+
+In order to avoid tedious and inefficient repetition of `seed!` all over the place we can instead define an instance of a `seed!` and pass it as a first argument of **either `rand` or `randn`**.
+
+```jl
+sco(
+"""
+my_seed = Random.seed!(123)
+"""
+)
+```
+
+
+```jl
+sco(
+"""
+rand(my_seed, 3)
+"""
+)
+```
+
+```jl
+sco(
+"""
+rand(my_seed, 3)
+"""
+)
+```
+
+### Downloads {#sec:downloads}
+
+One last thing from Julia's standard library for us to cover is the `Download` module.
+It will be really brief because we will only be covering a single function named `download`.
+
+Suppose you want to **download a file from the internet to your local storage**.
+You can accomplish this with the `download` function.
+The first and only required argument is the file's url.
+You can also specify as a second argument the desired output path for the downloaded file (don't forget the filesystem best practices!).
+If you don't specify a second argument, Julia will, by default, create a temporary file with the `tempfile` function.
+
+Let's import the `Download` module:
+
+```julia
+using Download
+```
+
+For example let's download our [`JuliaDataScience` GitHub repository](https://github.com/JuliaDataScience/JuliaDataScience) `Project.toml` file.
+Note that `download` function is not exported by `Downloads` module, so we have to use the `Module.function` syntax.
+By default it returns a string that holds the file path for the downloaded file:
+
+```jl
+scob(
+"""
+url = "https://raw.githubusercontent.com/JuliaDataScience/JuliaDataScience/main/Project.toml"
+
+my_file = Downloads.download(url) # tempfile() being created
+"""
+)
+```
+
+Let's just show the first 4 lines of our downloaded file with the `readlines` function:
+
+```jl
+sco(
+"""
+readlines(my_file)[1:4]
+"""
+)
+```
```

Comment:
@rikhuijzer Don't know to process this as a string... It looks really ugly.

Reply:
`sco("readlines(my_file)[1:4]"; process=string, post=output_block)`?

Reply:
Thanks